### ISSUE TITLE
Current state of ESCItem to be shown in the debugger

### ISSUE THREAD

**Issue by balloonpopper** [godot-escoria/escoria-issues#208]
**Is your feature request related to a problem? Please describe.**
If I run "set_state" I can't see what state my ESCItem is in when in the debugger

**Describe the solution you'd like**
I'd like to be able to click on an ESCItem and see what its "state" is

**Comment by dploeger** [comment]
The way to go would be checking out 

`escoria.object_manager.objects[global id of item].state`

I'd say that this should be clear for Godot developers and based on our [current documentation](https://docs.escoria-framework.org/en/devel/advanced/how_to_debug.html#escoria-data) and the [API docs](https://docs.escoria-framework.org/en/devel/api/ESCObjectManager.html).

Would you agree?

**Comment by balloonpopper** [comment]
Firstly, while the detail is hiding in the objects manager, I think it should be a field in the ESCItem so that I can quickly debug by finding the node in the node tree and looking at its member variables. It's not efficient or intuitive to have to click 
Node tree -> escoria -> Object Manager -> Find the right entry in the room objects array -> click your object -> look at the state

Secondly I **strongly** disagree that it's clear to developers. As you are one of the key developers (and maybe as you have a strong coding background?) it's likely to be obvious to you how the API documentation refers back to the codebase, but it's not from a new user's perspective. 

I look at a command like set_state and there's nothing there that immediately tells me to go looking in the object manager to find where the state is. Once you've used Escoria for a little while and you know that there _is_ an object manager, maybe the line "Changes the state of object" might give you a clue to go looking there, but you have to appreciate that a user who knows GDScript does not necessarily understand API documentation and how that relates to the code.

**Comment by dploeger** [comment]
Maybe you're right, that I'm assuming too much here, but I'd say if a developer starts debugging their game, they at least have a fair knowledge about Godot and GDScript (or else they wouldn't know about the debugging features at all) and we require this for advanced topics which debugging in my eyes clearly is.

Also we introduced the use of the API in the documentation stated above (which can be lacking currently of course) and we can't document *every* aspect of the API.

I still don't think we should add the state to the item, because in another PR we're debating whether to allow the same item in multiple rooms so a state is a room thing, not an item thing.

I'd like to know @StraToN and @BHSDuncan stance on this though.

**Comment by BHSDuncan** [comment]
There are valid points on both sides of this discussion. That said @dploeger would it be a lot of work to add this feature?

**Comment by dploeger** [comment]
Not really, we just need a new variable on ESCItem, that is set by ESCObjectsManager when the state changes. I just dislike the duplicate code and possible problem when an ESCItem is used twice...
